src/ui/QuerySummary/QuerySummary.ts

```diff
@@ -189,19 +241,101 @@ export class QuerySummary extends Component {
     }
   }
 
+  private parseNoResultsFoundMessage(noResultsFoundMessage: string) {
+    if (this.isQuerySummaryTagInMessage(noResultsFoundMessage)) {
+      let messageSections = noResultsFoundMessage.split(QUERY_TAG);
+      let messageElements = Array<Dom>();
+
+      _.each(messageSections, section => {
+        messageElements.push(this.createTextElement(section));
+      });
+
+      for (let i = 0, j = messageElements.length - 1; i < j; i++) {
+        if (messageElements[i].el.innerHTML != '' && messageElements[i + 1].el.innerHTML != '') {
+          messageElements.splice(i + 1, 0, this.getQueryElement());
+          i++;
+          j++;
+        }
+      }
+
+      _.each(messageElements, message => {
+        if (message.el.innerHTML == '') {
+          message = this.getQueryElement();
+        }
+      });
+
+      return messageElements;
+    } else {
+      return [this.createTextElement(noResultsFoundMessage)];
+    }
+  }
+
+  private isQuerySummaryTagInMessage(noResultsFoundMessage: string) {
+    return noResultsFoundMessage.split(QUERY_TAG).length > 0;
+  }
+
   private displayInfoOnNoResults() {
+    const noResultsFoundMessage = this.getNoResultsFoundMessageElement();
+    const cancelLastAction = this.getCancelLastActionElement();
+    const searchTipsInfo = this.getSearchTipsInfoElement();
+    const searchTips = this.getSearchTipsElement();
+
+    if (noResultsFoundMessage && this.options.enableNoResultsFoundMessage) {
+      this.textContainer.appendChild(noResultsFoundMessage.el);
+    }
+
+    if (this.options.enableCancelLastAction) {
+      this.textContainer.appendChild(cancelLastAction.el);
+    }
+
+    if (this.options.enableSearchTips) {
+      this.textContainer.appendChild(searchTipsInfo.el);
+      this.textContainer.appendChild(searchTips.el);
+    }
+  }
+
+  private getNoResultsFoundMessageElement() {
+    let parsedNoResultsFoundMessage = this.parseNoResultsFoundMessage(this.options.noResultsFoundMessage);
```

Comment:
`const` by default. Im going to stop adding this comment, but please change it for all the code you have modified.

Reply:
👍